mantidproject/mantid
# Translate sample transmission calculator into workbench

This PR adds the sample transmission calculator, originally only available in mantidplot, to workbench.

Report to: Stephen King

**To test:**

1. Build mantid and open `Interfaces->General->Sample Transmission Calculator`
2. Enter in some values for histogram bins and chemical cormula.
3. Click Calulate and check that the plot is updated with the correct output.
4. Try with invalid inputs and check that the validator allerts you that they are incorect.
1. Try in both single and multiple types for histogram and number and mass density.


Fixes #28691 


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
